src/dash/DashAdapter.js

```diff
@@ -914,28 +914,25 @@ function DashAdapter() {
                 }
 
                 // determine the relative insert position prior to possible removal
-                let relativePosition = (target[name + '_asArray'] || []).indexOf(leaf);
+                let relativePosition = (target[name] || []).indexOf(leaf);
                 let insertBefore = (operation.position === 'prepend' || operation.position === 'before');
 
                 // perform removal operation first, we have already capture the appropriate relative position
                 if (operation.action === 'remove' || operation.action === 'replace') {
                     // note that we ignore the 'ws' attribute of patch operations as it does not effect parsed mpd operations
 
                     // purge the directly named entity
-                    delete target[name];
-
-                    // if we did have a positional reference we need to purge from array set and restore X2JS proper semantics
-                    if (relativePosition != -1) {
-                        let targetArray = target[name + '_asArray'];
+                    if (typeof target[name] === Object) {
```

Comment:
```suggestion
                    if (typeof target[name] === 'object') {
```
`typeof` returns a string.

Reply:
Done in PR #4180